test/support/test_fixed_resource_policy_parallel_utils.h

```diff
@@ -0,0 +1,201 @@
+// -*- C++ -*-
+//===---------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_TEST_FIXED_RESOURCE_PARALLEL_UTILS_H
+#define _ONEDPL_TEST_FIXED_RESOURCE_PARALLEL_UTILS_H
+
+#include "support/test_config.h"
+#include "oneapi/dpl/dynamic_selection"
+#include "support/utils.h"
+#include <unordered_map>
+#include <thread>
+
+template <bool call_select_before_submit, typename Policy, typename UniverseContainer, typename UniverseMapping>
+int
+test_submit_and_wait(UniverseContainer u, UniverseMapping map, std::vector<int> actual, int offset=0)
+{
+    using my_policy_t = Policy;
+    my_policy_t p(u, offset);
+    std::vector<int> result(u.size(), 0);
+    const int N = 5;
+
+    std::vector<std::thread> threads;
+    std::mutex m;
+    auto func = [&result,&map, &m](typename oneapi::dpl::experimental::policy_traits<Policy>::resource_type e) {
+        int x = map[e];
+        result[x]++;
+
+        if constexpr (std::is_same_v<typename oneapi::dpl::experimental::policy_traits<Policy>::resource_type,
+                                     int>)
+            return e;
+        else
+            return typename oneapi::dpl::experimental::policy_traits<Policy>::wait_type{};
+    };
+    if(call_select_before_submit){
+        auto thread_func = [&p, &func, &m](){
+            for(int i=0;i<10;i++){
+                auto s = oneapi::dpl::experimental::select(p);
+                oneapi::dpl::experimental::submit_and_wait(s, func);
+
+            }
+        };
+
+        for(int i=0;i<5;i++){
+            threads.push_back(std::thread(thread_func));
+        }
+    }
+    else{
+        auto thread_func = [&p, &func, &m](){
+            for(int i=0;i<10;i++){
+                oneapi::dpl::experimental::submit_and_wait(p, func);
+            }
+        };
+
+        for(int i=0;i<5;i++){
+            threads.push_back(std::thread(thread_func));
+        }
+    }
+    for(auto& thread : threads){
+        thread.join();
+    }
+    bool pass = (actual == result);
+    if (!pass)
+    {
+        std::cout << "ERROR: did not select expected resources\n";
+        return 1;
+    }
+    std::cout << "submit_and_wait: OK\n";
+    return 0;
+}
+template <bool call_select_before_submit, typename Policy, typename UniverseContainer, typename UniverseMapping>
+int
+test_submit_and_wait_on_group(UniverseContainer u, UniverseMapping map, std::vector<int> actual, int offset=0)
+{
+    using my_policy_t = Policy;
+    my_policy_t p(u, offset);
+    std::vector<int> result(u.size(), 0);
+    const int N = 5;
+
+    std::mutex m;
```

Comment:
Which code we going to synchronize with this mutex?

Reply:
Unnecessary. Fixed.